From 1e38c0f1bb8fa2d8d7f819a42f6f76146af7c706 Mon Sep 17 00:00:00 2001
From: me-cafebabe <87921983+me-cafebabe@users.noreply.github.com>
Date: Tue, 8 Feb 2022 17:09:50 +0800
Subject: [PATCH 4/8] ak3: mi8937: add mi8937 devices names

---
 .ak3/anykernel.sh | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/.ak3/anykernel.sh b/.ak3/anykernel.sh
index bfee997..6706ad0 100755
--- a/.ak3/anykernel.sh
+++ b/.ak3/anykernel.sh
@@ -10,7 +10,17 @@ do.modules=0
 do.systemless=1
 do.cleanup=1
 do.cleanuponabort=0
-device.name1=REPLACE_DEVICE_NAME
+device.name1=mi8937
+device.name2=land
+device.name3=landtoni
+device.name4=riva
+device.name5=rolex
+device.name6=rova
+device.name7=santoni
+device.name8=ugg
+device.name9=ugglite
+device.name10=ulysse
+device.name11=prada
 supported.versions=REPLACE_ANDROID_VERSION
 supported.patchlevels=
 '; } # end properties
-- 
2.33.0

