From f8628edee78a8d7ea97b0555569e08d010db7941 Mon Sep 17 00:00:00 2001
From: me-cafebabe <87921983+me-cafebabe@users.noreply.github.com>
Date: Tue, 8 Feb 2022 23:55:47 +0800
Subject: [PATCH 6/8] ak3: mi8937-all: Combine dtb depending on the device

---
 .ak3/tools/ak3-core.sh | 46 ++++++++++++++++++++++++++++++++++++------
 1 file changed, 40 insertions(+), 6 deletions(-)

diff --git a/.ak3/tools/ak3-core.sh b/.ak3/tools/ak3-core.sh
index fb31921..18a05da 100755
--- a/.ak3/tools/ak3-core.sh
+++ b/.ak3/tools/ak3-core.sh
@@ -253,12 +253,46 @@ flash_boot() {
   done;
 
   cd $home;
-  for i in zImage zImage-dtb Image Image-dtb Image.gz Image.gz-dtb Image.bz2 Image.bz2-dtb Image.lzo Image.lzo-dtb Image.lzma Image.lzma-dtb Image.xz Image.xz-dtb Image.lz4 Image.lz4-dtb Image.fit; do
-    if [ -f $i ]; then
-      kernel=$home/$i;
-      break;
-    fi;
-  done;
+#  for i in zImage zImage-dtb Image Image-dtb Image.gz Image.gz-dtb Image.bz2 Image.bz2-dtb Image.lzo Image.lzo-dtb Image.lzma Image.lzma-dtb Image.xz Image.xz-dtb Image.lz4 Image.lz4-dtb Image.fit; do
+#    if [ -f $i ]; then
+#      kernel=$home/$i;
+#      break;
+#    fi;
+#  done;
+  case "$(cat /sys/firmware/devicetree/base/model)" in
+    "Qualcomm Technologies, Inc. MSM8917-PMI8937 QRD SKU5")
+      if [ -e "/sys/class/leds/infrared/transmit" ]; then
+        DTB_FILE="msm8917-pmi8937-qrd-sku5-rolex.dtb"
+      elif [ -e "/dev/spidev7.1" ]; then
+        DTB_FILE="msm8917-pmi8937-qrd-sku5-riva.dtb"
+      else
+        DTB_FILE="msm8917-pmi8937-qrd-sku5-tiare.dtb"
+      fi
+      ;;
+    "Qualcomm Technologies, Inc. MSM8917 QRD SKU5")
+      DTB_FILE="msm8917-pmi8937-qrd-sku5-riva-Oreo.dtb"
+      ;;
+    "Qualcomm Technologies, Inc. MSM8917-PMI8937 MTP")
+      DTB_FILE="msm8917-pmi8937-mtp-ugglite.dtb"
+      ;;
+    "Qualcomm Technologies, Inc. MSM8940-PMI8937 MTP")
+      DTB_FILE="msm8940-pmi8937-mtp-ugg.dtb"
+      ;;
+    "Qualcomm Technologies, Inc. MSM8937-PMI8950 QRD SKU1")
+      DTB_FILE="msm8937-pmi8950-qrd-land.dtb"
+      ;;
+    "Qualcomm Technologies, Inc. MSM8940-PMI8950 QRD SKU7")
+      DTB_FILE="msm8940-pmi8950-qrd-sku7-santoni.dtb"
+      ;;
+    "Qualcomm Technologies, Inc. MSM8937-PMI8950 MTP")
+      DTB_FILE="msm8937-pmi8950-mtp-prada.dtb"
+      ;;
+    *)
+      abort "Fatal Error: Unable to determine device from devicetree!"
+      ;;
+  esac
+  cat Image.gz dtbs/$DTB_FILE > Image.gz-dtb
+  kernel=$home/Image.gz-dtb
   if [ "$kernel" ]; then
     if [ -f "$bin/mkmtkhdr" -a -f "$split_img/boot.img-base" ]; then
       $bin/mkmtkhdr --kernel $kernel;
-- 
2.33.0

